Update meme tools to version 4.11.1.0

gregvonkuster opened the pull request:
to depend on packages that are conda-compatible

gregvonkuster commented:
I'm not an expert on the underlying testing environment for this, so I'm not quite sure if I need to fix anything in the tools to correct these tests or if the test datasets are causing the problem.  These tests passed when not using conda resolution for underlying requirements.

  Any help is much appreciated!  The errors look like this:
```
| 5 - MEME on data 2 and data 1 (xml) (HID - NAME) 
| Dataset Blurb:
|  error
| Dataset Info:
|  discarding /home/travis/conda/envs/iuc_conda/bin from PATH
|  prepending /tmp/tmpaa2oy6/job_working_directory/000/14/conda-env/bin to PATH
|  Duplicate sequence name. Skipping 'chr21_47517957_47518007_+'.
|  Duplicate sequence name. Skipping 'chr21_47517957_47518
| Dataset Job Standard Output:
|  discarding /home/travis/conda/envs/iuc_conda/bin from PATH
|  prepending /tmp/tmpaa2oy6/job_working_directory/000/14/conda-env/bin to PATH
|  Duplicate sequence name. Skipping 'chr21_47517957_47518007_+'.
|  Duplicate sequence name. Skipping 'chr21_47517957_47518007_+'.
|  Duplicate sequence name. Skipping 'chr21_47575506_47575556_-'.
|  Error in: ceqlogo.c
|  Line: 250
|  SYSTEM ERROR:
|  Cannot open file '/anaconda/envs/_build/etc/template.eps'
|  (line: 1108, file: ceqlogo.c)
|  Error running MEME.
| Dataset Job Standard Error:
|  mv: cannot stat `/tmp/tmpaa2oy6/job_working_directory/000/14/dataset_28_files/meme.html': No such file or directory
```
Also, it is not related to the error above, but in my local development / test environment, I get this error.  This is after I followed the steps outlined here: https://github.com/galaxyproject/tools-iuc/blob/master/.travis.yml.  I completely removed the miniconda environment I had and installed everything from scratch.
```
$ planemo test --conda_dependency_resolution --conda_prefix "$CONDA_PREFIX" --galaxy_branch $GALAXY_RELEASE --galaxy_source $GALAXY_REPO .
git --git-dir /home/greg/.planemo/gx_repo fetch >/dev/null 2>&1
cd /tmp/tmp1InXiP; git clone  --branch 'release_16.04' '/home/greg/.planemo/gx_repo' 'galaxy-dev'; cd galaxy-dev; if [ -d .venv ] || [ -f dist-eggs.ini ]; then GALAXY_VIRTUAL_ENV=.venv;  else GALAXY_VIRTUAL_ENV=/home/greg/.planemo/gx_venv; fi; export GALAXY_VIRTUAL_ENV; if [ ! -e $GALAXY_VIRTUAL_ENV ]; then /home/greg/miniconda2/bin/virtualenv -p /home/greg/conda/bin/python2.7 $GALAXY_VIRTUAL_ENV; fi; [ -e $GALAXY_VIRTUAL_ENV ] && . $GALAXY_VIRTUAL_ENV/bin/activate; COMMON_STARTUP_ARGS=; $(grep -q 'skip-venv' run_tests.sh) && COMMON_STARTUP_ARGS="--dev-wheels"; export COMMON_STARTUP_ARGS; echo "Set COMMON_STARTUP_ARGS to ${COMMON_STARTUP_ARGS}"; ./scripts/common_startup.sh ${COMMON_STARTUP_ARGS}
Cloning into 'galaxy-dev'...
done.
/bin/sh: /home/greg/.planemo/gx_venv/bin/activate: No such file or directory
Testing using galaxy_root /tmp/tmp1InXiP/galaxy-dev
Testing tools with command [cd /tmp/tmp1InXiP/galaxy-dev; if [ -d .venv ] || [ -f dist-eggs.ini ]; then GALAXY_VIRTUAL_ENV=.venv;  else GALAXY_VIRTUAL_ENV=/home/greg/.planemo/gx_venv; fi; export GALAXY_VIRTUAL_ENV; if [ ! -e $GALAXY_VIRTUAL_ENV ]; then /home/greg/miniconda2/bin/virtualenv -p /home/greg/conda/bin/python2.7 $GALAXY_VIRTUAL_ENV; fi; [ -e $GALAXY_VIRTUAL_ENV ] && . $GALAXY_VIRTUAL_ENV/bin/activate; sh run_tests.sh $COMMON_STARTUP_ARGS --report_file /home/greg/work/tools-iuc/meme_4_11_1_0/tools-iuc/tools/meme/tool_test_output.html --xunit_report_file /tmp/tmp1InXiP/xunit.xml --structured_data_report_file /home/greg/work/tools-iuc/meme_4_11_1_0/tools-iuc/tools/meme/tool_test_output.json functional.test_toolbox]
/bin/sh: /home/greg/.planemo/gx_venv/bin/activate: No such file or directory
Cannot locate xUnit report [/tmp/tmp1InXiP/xunit.xml] for tests - required to build planemo report and summarize tests.
Traceback (most recent call last):
  File "/home/greg/miniconda2/bin/planemo", line 11, in <module>
    sys.exit(planemo())
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args[1:], **kwargs)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/planemo/cli.py", line 174, in handle_profile_options
    return f(*args, **kwds)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/planemo/commands/cmd_test.py", line 78, in cli
    return_value = run_in_config(ctx, config, **kwds)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/planemo/galaxy/test/actions.py", line 100, in run_in_config
    _check_test_outputs(xunit_report_file_tracker, structured_report_file_tracker)
  File "/home/greg/miniconda2/lib/python2.7/site-packages/planemo/galaxy/test/actions.py", line 281, in _check_test_outputs
    raise Exception(message)
Exception: Cannot locate xUnit report [/tmp/tmp1InXiP/xunit.xml] for tests - required to build planemo report and summarize tests.
```

bgruening commented:
The first error indicates a problem with the conda package. It seems some files are missing under /etc/.
The second error indicates that conda could not be activated `/home/greg/.planemo/gx_venv/bin/activate` which results in no dependency and non-passing tests. The planemo error message is confusing and wrong you will find the real error above.

gregvonkuster commented:
@bgruening Thanks for the help.
Regarding the 2nd error, this is what get placed in my environment when I install everything from scratch.
```
$ pwd
/home/greg/.planemo/gx_venv/bin
$ ll
total 20
drwxrwxr-x. 2 greg greg  4096 Jul  5 13:39 .
drwxrwxr-x. 5 greg greg  4096 Jul  5 13:39 ..
lrwxrwxrwx. 1 greg greg     9 Jul  5 13:39 python -> python2.7
lrwxrwxrwx. 1 greg greg     9 Jul  5 13:39 python2 -> python2.7
-rwxrwxr-x. 1 greg greg 10038 Jul  5 13:39 python2.7
```
So I must have to do something in addition to what is listed in this script: https://github.com/galaxyproject/tools-iuc/blob/master/.travis.yml

But I'm not quite sure what it is.

gregvonkuster commented:
BTW, I'm using miniconda if that matters.   Should I be using standard conda?

nsoranzo commented:
@gregvonkuster Miniconda should be fine.

gregvonkuster commented:
I figured out what was causing the problem in my local environment that was not allowing planemo to test.  Installing miniconda prepends the conda location to the PATH environment variable rather than appending it - in my .bashrc I have:
```
# added by Miniconda2 4.0.5 installer
export PATH="/home/greg/miniconda2/bin:$PATH"
```
I had to set my $PATH to locate the system Python before locating conda and things worked.

nsoranzo commented:
Fimo tests are passing, for meme it seems that the first command fails. I'd suggest to remove 
```
|| echo "Error running MEME."
```
from `<command>` to prevent the `mv` to run and fail (this will not obviously fix the problem).

gregvonkuster commented:
Thanks @nsoranzo 
Here is the error which is logged in the job's stderr - I'm not quite sure what might be causing it.  Any ideas?  A file named prior30.plib is 1 of the data files used by 1 of the 2 functional tests for meme, but it is unclear why meme is looking for a file by this name in the conda environment
```
Can't find prior library /anaconda/envs/_build/etc/prior30.plib
```
Here is the entire test output for the failed meme tests.
```
  File "/usr/lib64/python2.7/unittest/case.py", line 369, in run
    testMethod()
  File "/tmp/tmpHpq7JS/galaxy-dev/test/functional/test_toolbox.py", line 299, in test_tool
    self.do_it( td )
  File "/tmp/tmpHpq7JS/galaxy-dev/test/functional/test_toolbox.py", line 75, in do_it
    raise e
'Job in error state.\nJob in error state.\nJob in error state.\nJob in error state.\n-------------------- >> begin captured stdout << ---------------------\nHistory with id 8155e4b4bf1581ff in error - summary of datasets in error below.\n--------------------------------------\n| 2 - MEME on data 1 (html) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 3 - MEME on data 1 (text) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 4 - MEME on data 1 (xml) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\nHistory with id 8155e4b4bf1581ff in error - summary of datasets in error below.\n--------------------------------------\n| 2 - MEME on data 1 (html) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 3 - MEME on data 1 (text) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 4 - MEME on data 1 (xml) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\nHistory with id 8155e4b4bf1581ff in error - summary of datasets in error below.\n--------------------------------------\n| 2 - MEME on data 1 (html) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 3 - MEME on data 1 (text) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 4 - MEME on data 1 (xml) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\nHistory with id 8155e4b4bf1581ff in error - summary of datasets in error below.\n--------------------------------------\n| 2 - MEME on data 1 (html) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 3 - MEME on data 1 (text) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n| 4 - MEME on data 1 (xml) (HID - NAME) \n| Dataset Blurb:\n|  error\n| Dataset Info:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_4751800\n| Dataset Job Standard Output:\n|  discarding /home/greg/conda/envs/iuc_conda/bin from PATH\n|  prepending /tmp/tmpHpq7JS/job_working_directory/000/11/conda-env/bin to PATH\n| Dataset Job Standard Error:\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\n|  Duplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\n|  Can\'t find prior library /anaconda/envs/_build/etc/prior30.plib\n|\n--------------------------------------\n\n--------------------- >> end captured stdout << ----------------------\n-------------------- >> begin captured logging << --------------------\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/users?key=test_key HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.web.framework.webapp: INFO: Session authenticated using Galaxy master api key\nrequests.packages.urllib3.connectionpool: DEBUG: "POST /api/users/2891970512fa2d5a/api_key HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "POST /api/histories HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.tools: DEBUG: Validated and populated state for tool request (40.855 ms)\ngalaxy.tools.actions.upload: DEBUG: Persisted uploads (17.236 ms)\ngalaxy.tools.actions.upload: DEBUG: Checked and cleaned uploads (110.295 ms)\ngalaxy.tools.actions.upload_common: INFO: tool upload1 created job id 10\ngalaxy.tools.actions.upload: DEBUG: Created upload job (80.829 ms)\ngalaxy.tools.execute: DEBUG: Tool [upload1] created job [10] (208.862 ms)\ngalaxy.tools.execute: DEBUG: Executed 1 job(s) for tool upload1 request: (243.153 ms)\nrequests.packages.urllib3.connectionpool: DEBUG: "POST /api/tools HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs: DEBUG: (10) Working directory for job is: /tmp/tmpHpq7JS/job_working_directory/000/10\ngalaxy.jobs.handler: DEBUG: (10) Dispatching to planemo_runner runner\ngalaxy.jobs: DEBUG: (10) Persisting job destination (destination id: planemo_dest)\ngalaxy.jobs.runners: DEBUG: Job [10] queued (62.897 ms)\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs.handler: INFO: (10) Job dispatched\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.tools.deps: DEBUG: Building dependency shell command for dependency \'samtools\'\ngalaxy.tools.deps: DEBUG: Find dependency samtools version None\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: WARNING: Failed to resolve dependency on \'samtools\', ignoring\ngalaxy.jobs.command_factory: INFO: Built script [/tmp/tmpHpq7JS/job_working_directory/000/10/tool_script.sh] for tool command[python /tmp/tmpHpq7JS/galaxy-dev/tools/data_source/upload.py /tmp/tmpHpq7JS/galaxy-dev /tmp/tmpHpq7JS/tmp/tmpWbtqcw /tmp/tmpHpq7JS/tmp/tmp7O3Qsh 22:/tmp/tmpHpq7JS/job_working_directory/000/10/dataset_22_files:/tmp/tmpHpq7JS/files/000/dataset_22.dat]\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.tools.deps: DEBUG: Building dependency shell command for dependency \'samtools\'\ngalaxy.tools.deps: DEBUG: Find dependency samtools version None\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: WARNING: Failed to resolve dependency on \'samtools\', ignoring\ngalaxy.jobs.runners: DEBUG: (10) command is: mkdir -p working; cd working; /tmp/tmpHpq7JS/job_working_directory/000/10/tool_script.sh; return_code=$?; cd \'/tmp/tmpHpq7JS/job_working_directory/000/10\'; python "/tmp/tmpHpq7JS/job_working_directory/000/10/set_metadata_8uUf8B.py" "/tmp/tmpHpq7JS/tmp/tmpWbtqcw" "/tmp/tmpHpq7JS/job_working_directory/000/10/working/galaxy.json" "/tmp/tmpHpq7JS/job_working_directory/000/10/metadata_in_HistoryDatasetAssociation_22_jD4VH9,/tmp/tmpHpq7JS/job_working_directory/000/10/metadata_kwds_HistoryDatasetAssociation_22_wULDmn,/tmp/tmpHpq7JS/job_working_directory/000/10/metadata_out_HistoryDatasetAssociation_22_3Orx7t,/tmp/tmpHpq7JS/job_working_directory/000/10/metadata_results_HistoryDatasetAssociation_22_EdOX0_,/tmp/tmpHpq7JS/files/000/dataset_22.dat,/tmp/tmpHpq7JS/job_working_directory/000/10/metadata_override_HistoryDatasetAssociation_22_2YUEVH" 5242880; sh -c "exit $return_code"\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs.runners.local: DEBUG: (10) executing job script: /tmp/tmpHpq7JS/job_working_directory/000/10/galaxy_10.sh\ngalaxy.jobs: DEBUG: (10) Persisting job destination (destination id: planemo_dest)\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs.runners.local: DEBUG: execution finished: /tmp/tmpHpq7JS/job_working_directory/000/10/galaxy_10.sh\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.model.metadata: DEBUG: loading metadata from file for: HistoryDatasetAssociation 22\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs: DEBUG: job 10 ended (finish() executed in (406.996 ms))\ngalaxy.tools: DEBUG: Validated and populated state for tool request (46.117 ms)\ngalaxy.tools.actions: INFO: Handled output named html_outfile for tool meme_meme (24.625 ms)\ngalaxy.tools.actions: INFO: Handled output named txt_outfile for tool meme_meme (18.927 ms)\ngalaxy.tools.actions: INFO: Handled output named xml_outfile for tool meme_meme (18.707 ms)\ngalaxy.tools.actions: INFO: Added output datasets to history (13.826 ms)\ngalaxy.tools.actions: INFO: Verified access to datasets for Job[unflushed,tool_id=meme_meme] (10.223 ms)\ngalaxy.tools.actions: INFO: Setup for job Job[unflushed,tool_id=meme_meme] complete, ready to flush (22.477 ms)\ngalaxy.tools.actions: INFO: Flushed transaction for job Job[id=11,tool_id=meme_meme] (19.887 ms)\ngalaxy.tools.execute: DEBUG: Tool [meme_meme] created job [11] (137.836 ms)\ngalaxy.tools.execute: DEBUG: Executed 1 job(s) for tool meme_meme request: (182.572 ms)\nrequests.packages.urllib3.connectionpool: DEBUG: "POST /api/tools HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs: DEBUG: (11) Working directory for job is: /tmp/tmpHpq7JS/job_working_directory/000/11\ngalaxy.jobs.handler: DEBUG: (11) Dispatching to planemo_runner runner\ngalaxy.jobs: DEBUG: (11) Persisting job destination (destination id: planemo_dest)\ngalaxy.jobs.runners: DEBUG: Job [11] queued (92.498 ms)\ngalaxy.jobs.handler: INFO: (11) Job dispatched\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\ngalaxy.tools.deps: DEBUG: Building dependency shell command for dependency \'graphicsmagick\'\ngalaxy.tools.deps: DEBUG: Find dependency graphicsmagick version 1.3.20\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.conda.CondaDepenency object at 0x7f4494655990> (isnull? False)\ngalaxy.tools.deps: DEBUG: Building dependency shell command for dependency \'meme\'\ngalaxy.tools.deps: DEBUG: Find dependency meme version 4.11.1\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.conda.CondaDepenency object at 0x7f4494655590> (isnull? False)\ngalaxy.jobs.command_factory: INFO: Built script [/tmp/tmpHpq7JS/job_working_directory/000/11/tool_script.sh] for tool command[[ "$CONDA_DEFAULT_ENV" = "/tmp/tmpHpq7JS/job_working_directory/000/11/conda-env" ] || . /home/greg/conda/bin/activate \'/tmp/tmpHpq7JS/job_working_directory/000/11/conda-env\' 2>&1 ; [ "$CONDA_DEFAULT_ENV" = "/tmp/tmpHpq7JS/job_working_directory/000/11/conda-env" ] || . /home/greg/conda/bin/activate \'/tmp/tmpHpq7JS/job_working_directory/000/11/conda-env\' 2>&1 ; meme "/tmp/tmpHpq7JS/files/000/dataset_22.dat" -o "/tmp/tmpHpq7JS/job_working_directory/000/11/dataset_23_files" -nostatus -maxsize 1000000 && mv /tmp/tmpHpq7JS/job_working_directory/000/11/dataset_23_files/meme.html /tmp/tmpHpq7JS/files/000/dataset_23.dat && mv /tmp/tmpHpq7JS/job_working_directory/000/11/dataset_23_files/meme.txt /tmp/tmpHpq7JS/files/000/dataset_24.dat && mv /tmp/tmpHpq7JS/job_working_directory/000/11/dataset_23_files/meme.xml /tmp/tmpHpq7JS/files/000/dataset_25.dat]\ngalaxy.tools.deps: DEBUG: Building dependency shell command for dependency \'samtools\'\ngalaxy.tools.deps: DEBUG: Find dependency samtools version None\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: DEBUG: Resolver conda returned <galaxy.tools.deps.resolvers.NullDependency object at 0x4573e90> (isnull? True)\ngalaxy.tools.deps: WARNING: Failed to resolve dependency on \'samtools\', ignoring\ngalaxy.jobs.runners: DEBUG: (11) command is: mkdir -p working; cd working; /tmp/tmpHpq7JS/job_working_directory/000/11/tool_script.sh; return_code=$?; cd \'/tmp/tmpHpq7JS/job_working_directory/000/11\'; python "/tmp/tmpHpq7JS/job_working_directory/000/11/set_metadata_TDxWhV.py" "/tmp/tmpHpq7JS/tmp/tmpWbtqcw" "/tmp/tmpHpq7JS/job_working_directory/000/11/working/galaxy.json" "/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_in_HistoryDatasetAssociation_23_txuXPD,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_kwds_HistoryDatasetAssociation_23_L2ujUz,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_out_HistoryDatasetAssociation_23_jY5xZO,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_results_HistoryDatasetAssociation_23_cIKHAx,/tmp/tmpHpq7JS/files/000/dataset_23.dat,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_override_HistoryDatasetAssociation_23_jPtfUC" "/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_in_HistoryDatasetAssociation_24_KTxCx6,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_kwds_HistoryDatasetAssociation_24_uO6Pp2,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_out_HistoryDatasetAssociation_24_QoCLJk,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_results_HistoryDatasetAssociation_24_aylXXC,/tmp/tmpHpq7JS/files/000/dataset_24.dat,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_override_HistoryDatasetAssociation_24_aRcHms" "/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_in_HistoryDatasetAssociation_25_yMvb_q,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_kwds_HistoryDatasetAssociation_25_LR8ZAU,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_out_HistoryDatasetAssociation_25_aUx3sB,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_results_HistoryDatasetAssociation_25_krjPS4,/tmp/tmpHpq7JS/files/000/dataset_25.dat,/tmp/tmpHpq7JS/job_working_directory/000/11/metadata_override_HistoryDatasetAssociation_25_RIdWn9" 5242880; sh -c "exit $return_code"\ngalaxy.jobs.runners.local: DEBUG: (11) executing job script: /tmp/tmpHpq7JS/job_working_directory/000/11/galaxy_11.sh\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\ngalaxy.jobs: DEBUG: (11) Persisting job destination (destination id: planemo_dest)\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\ngalaxy.jobs.runners.local: DEBUG: execution finished: /tmp/tmpHpq7JS/job_working_directory/000/11/galaxy_11.sh\ngalaxy.jobs.output_checker: DEBUG: Tool produced standard error failing job - [Duplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\nDuplicate sequence name. Skipping \'chr21_47517957_47518007_+\'.\nDuplicate sequence name. Skipping \'chr21_47575506_47575556_-\'.\nCan\'t find prior library /anaconda/envs/_build/etc/prior30.pl]\ngalaxy.jobs: DEBUG: (11) setting dataset 23 state to ERROR\ngalaxy.jobs: DEBUG: (11) setting dataset 24 state to ERROR\ngalaxy.jobs: DEBUG: (11) setting dataset 25 state to ERROR\ngalaxy.jobs: DEBUG: job 11 ended (finish() executed in (286.420 ms))\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?full=true&key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/jobs/9ce08b2254e4d5ed?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/3e4e8de146a035f5/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/a1e670e1ea0da54b/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\nrequests.packages.urllib3.connectionpool: INFO: Starting new HTTP connection (1): localhost\nrequests.packages.urllib3.connectionpool: DEBUG: "GET /api/histories/8155e4b4bf1581ff/contents/9a22496ee796fbfc/provenance?key=ec4e0c53000701958da0ef6a158a89ba HTTP/1.1" 200 None\n--------------------- >> end captured logging << ---------------------'

meme_meme (Test #2, Failed)

```

gregvonkuster commented:
@bgruening I'll need some additional help to get the underlying bioconda recipe for meme fixed so that this version of the meme tool passes functional tests.  ;)

I've got a new version of this meme recipe that passes conda testing, but how can I use it when testing this new version of the meme tool to see if the changes I've made to the recipe will correct this problem?  I assume I have to somehow override the bioconda channel that is pulling in the meme recipe.

If I have my newer version of the meme recipe that I tested from within its directory using this command:
```
conda build . --channel bioconda
```
The test results were generated here:
```
/home/greg/miniconda2/envs/_test
```
The contents of that directory look ok:
```
drwxrwxr-x.  2 greg greg 12288 Jul  8 08:53 bin
drwxrwxr-x.  2 greg greg  4096 Jul  8 08:53 conda-meta
drwxrwxr-x.  6 greg greg  4096 Jul  8 08:53 doc
drwxrwxr-x.  2 greg greg  4096 Jul  8 08:53 etc
drwxrwxr-x.  5 greg greg  4096 Jul  8 08:53 gcc
drwxrwxr-x. 11 greg greg  4096 Jul  8 08:53 include
drwxrwxr-x. 16 greg greg 12288 Jul  8 08:53 lib
drwxrwxr-x. 12 greg greg  4096 Jul  8 08:53 share
drwxrwxr-x.  3 greg greg  4096 Jul  8 08:53 ssl
```
And the problematic prior30.plib file is in the etc directory.  So how can I use it when executing planemo to test the meme tool?  I'm not really sure where to start trying to fix the recipe that produces this error with planemo testing the meme tool, so I'm kind of bungling around here.
```
Can't find prior library /anaconda/envs/_build/etc/prior30.plib
```
Thanks for the help!  ;)

nsoranzo commented:
@gregvonkuster I'm in the process of submitting a PR to bioconda, I think I've fixed the problem.

gregvonkuster commented:
ok, thanks @nsoranzo

gregvonkuster commented:
@nsoranzo Can you please add this patch to your PR:
```
--- src/display.c
+++ src/display.c
@@ -1409,7 +1409,7 @@
   if (dataset->evt == BIG) {
     strcpy(evt_string, "inf");
   } else {
-    sprintf(evt_string, "%8g", dataset->evt);
+    sprintf(evt_string, "%g", dataset->evt);
   }
 
   fprintf(
```

nsoranzo commented:
@gregvonkuster Ok, can you try to get this applied upstream?

nsoranzo commented:
@gregvonkuster Actually, that's already been applied upstream and it is not needed anymore in meme 4.11.1 .

nsoranzo commented:
@gregvonkuster See https://github.com/bioconda/bioconda-recipes/pull/1938 for the meme bioconda recipe fix.

gregvonkuster commented:
Thanks @nsoranzo!  ;)

nsoranzo commented:
Closing in favour of #871.
